src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala

```diff
@@ -392,11 +392,25 @@ package object metadata extends DebugEnhancedLogging {
 
   def filesXmlFilesHaveOnlyAllowedAccessRights(t: TargetBag): Try[Unit] = {
     trace(())
-    t.tryFilesXml.map { xml =>
-      val accessRights = xml \ "file" \ "accessRights"
-      val hasOnlyAllowedAccessRights = accessRights.forall(r => allowedAccessRights.contains(r.text))
-      if (!hasOnlyAllowedAccessRights) fail("files.xml: invalid access right(s) in dct:accessRights element (allowed values ANONYMOUS, RESTRICTED_REQUEST and NONE")
+      for {
+        xml <- t.tryFilesXml
+        accessRights <- getAccessRightElements(xml)
+        _ <- accessRights.map(validateAccessRights).collectResults.recoverWith {
+          case ce: CompositeException => Try(fail(ce.getMessage))
+        }
+      } yield ()
     }
+
+  private def getAccessRightElements(xml: Elem): Try[Seq[NodeSeq]] = Try {
+    trace(())
+    val accessRights = xml \ "file" \ "accessRights"
+    val accessibleToRights = xml \ "file" \ "accessibleToRights"
+    val visibleToRights = xml \ "file" \ "visibleToRights"
+    Seq(accessRights, accessibleToRights, visibleToRights)
```

Comment:
It was not clear to me whether also accessibleToRights and visibleToRights should be validated, or whether their values have been already validated elsewhere.  If not needed, we can easily remove these elements from the sequence of elements that are validated.

Reply:
accessibleToRights and visibleToRights are limited in values in the files.xsd. for accessRights that was not possible, as it is in a different namespace